src/python/WMCore/MicroService/DataStructs/Workflow.py

```diff
@@ -347,9 +352,14 @@ def getChunkBlocks(self, numChunks=1):
                 if not sortedPrimary or idx >= len(sortedPrimary):
                     # then all blocks have been distributed
                     break
-                elif thisChunkSize + sortedPrimary[idx][1] <= chunkSize:
+                elif not thisChunkSize:
+                    # then this site/chunk is empty, assign a block to it
                     thisChunk.add(sortedPrimary[idx][0])
-                    thisChunkSize += sortedPrimary[idx][1]
+                    thisChunkSize += sortedPrimary[idx][1]['blockSize']
+                    sortedPrimary.pop(idx)
+                elif thisChunkSize + sortedPrimary[idx][1]['blockSize'] <= chunkSize:
```

Comment:
IIUC here you are iterating through the sortedPrimary list (until it's exhausted) but you are making the checks towards the chunkSize - basically you check how well you are filling a second list. Did you check if under some strange condition with some weird block size distribution, we do not end up with some leftovers - I mean not being able to fit in all the blocks into the available chunks?

Reply:
Yes, I check against the target chunk size (`chunkSize`), while we find blocks that fit in there, we append blocks to the same site.

About the leftovers, that's a good point and I did foresee such cases. Logic is a few lines below (# now take care of the leftovers...)